UserInterface.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Weather_Monitoring_System.DataParsers;
+
+namespace Weather_Monitoring_System
+{
+    public class UserInterface
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Welcom To Weather Monitoring System");
+            Console.WriteLine("Enter the weather data usign JSON or XML format\n");
+
+            var weatherDataInput = Console.ReadLine();
+
+            if (weatherDataInput.StartsWith('{'))
```

Comment:
You are supposed to use some design pattern like the ones mentioned in the ticket to do this job (determining the parser).
Also, this is not something that should be here in the UI service.

Reply:
Can you explain this more, please?

Reply:
The task was added under the SOLID & Design Principles topic so you are supposed to expose what you have learned from these topics in coding the task. 
Please read the task carefully and see what they request. You should at least use the Strategy and Observer design patterns. 
Try to think about how to use them in the context of this task.

Reply:
I am still thinking about this....., I will push some changes but still, I haven't fixed this, I will think more and try to fix the code according to this